### PR TITLE
[A2-771] follow-up ProjectsAuthorized fix

### PR DESCRIPTION
**Pull request by bcmdarroch**
### :nut_and_bolt: Description
follow-up to https://github.com/chef/automate/pull/135 

### :+1: Definition of Done
- [x] when all projects are allowed and the request contains no project headers, authz-service passes on `*` to the domain services

### :athletic_shoe: Demo Script / Repro Steps
```
[16][default:/src:0]# chef-automate iam upgrade-to-v2 --beta2.1

Enabling IAM v2.1
Migrating v1 policies...
Creating default teams Editors and Viewers...

Migrating existing teams...

Success: Enabled IAM v2.1

[22][default:/src:0]# curl -H "api-token: $TOK"  -d '{"id": "p1", "name": "project 1"}' https://localhost/apis/iam/v2beta/projects --insecure | jq
{
  "project": {
    "name": "project 1",
    "id": "p1",
    "type": "CUSTOM",
    "projects": [
      "p1"
    ]
  }
}

[18][default:/src:0]# curl -H "api-token: $TOK"  -d '{"id": "team1", "name": "team 1", "projects": ["p1"]}' https://localhost/apis/iam/v2beta/teams --insecure | jq
{
  "team": {
    "id": "team1",
    "name": "team 1",
    "projects": [
      "p1"
    ]
  }
}

[23][default:/src:0]# curl -H "api-token: $TOK"  https://localhost/apis/iam/v2beta/teams --insecure | jq
{
  "teams": [
    {
      "id": "admins",
      "name": "Members of the admins team, by default, have access to all parts of the API.",
      "projects": []
    },
    {
      "id": "editors",
      "name": "Editors",
      "projects": []
    },
    {
      "id": "viewers",
      "name": "Viewers",
      "projects": []
    },
    {
      "id": "team1",
      "name": "team 1",
      "projects": [
        "p1"
      ]
    }
  ]
}

[26][default:/src:0]# sl
--> Tailing the Habitat Supervisor's output (use 'Ctrl+c' to stop)
...
[2019-04-30T20:33:48+00:00] authz-service.default(O): time="2019-04-30T20:33:48Z" level=info msg="Projects Authorized Query" action="iam:teams:list" projects="[]" resource="iam:teams" result="[*]" subject="[token:e0a18396-7960-4a12-a185-3ef02f6df5f3]"
```
(punch line is `result=[*]` in the log)

### :white_check_mark: Checklist

- [ ] Necessary tests added/updated?
- [ ] Necessary docs added/updated?
- [ ] Code actually executed?
- [ ] Vetting performed (unit tests, lint, etc.)?

### CODE REVIEW

File: components/authz-service/server/v2/authz.go [hunk 1]
```diff
@@ -89,12 +89,16 @@ func (s *authzServer) ProjectsAuthorized(
 	}
 
 	requestedProjects := req.ProjectsFilter
+
+	allProjectsRequested := len(req.ProjectsFilter) == 0
```
Comment (tylercloke) [inline components/authz-service/server/v2/authz.go]: Nice cleanup

File: components/authz-service/server/v2/system.go [hunk 2]
```diff
@@ -171,6 +171,8 @@ func SystemPolicies() []*storage.Policy {
 					"iam:roles:" + constants.EditorRoleID,
 					"iam:roles:" + constants.ViewerRoleID,
 					"iam:roles:" + constants.IngestRoleID,
+					"iam:roles:" + constants.ProjectAdminRoleID,
+					"iam:roles:" + constants.IAMMembersViewerRoleID,
```
Comment (msorens) [inline components/authz-service/server/v2/system.go]: Good catch... but unrelated to this PR, is that correct...?

File: components/authz-service/server/v2/system.go [hunk 3]
```diff
@@ -185,7 +185,6 @@ func SystemPolicies() []*storage.Policy {
 				},
 				Projects: []string{constants.AllProjectsID},
 			},
-			// TODO can't delete admin team from admin policy
```
Comment (msorens) [inline components/authz-service/server/v2/system.go]: By removing this comment you are claiming that it is now true that you cannot delete the admin team, but is that true now...?

File: components/authz-service/engine/conformance/conformance_v2_test.go [hunk 4]
```diff
@@ -82,18 +82,16 @@ func TestV2IsAuthorized(t *testing.T) {
 
 func TestV2ProjectsAuthorized(t *testing.T) {
 	ctx, engines := setup(t)
-	sub, act, res, proj1, proj2 := "user:local:admin", "iam:users:create", "iam:users", "proj-1", "proj-2"
+	sub, act, res := "user:local:admin", "iam:users:create", "iam:users"
+	proj1, proj2, proj3, proj4, unassigned := "proj-1", "proj-2", "proj-3", "proj-4", "(unassigned)"
```
Comment (msorens) [inline components/authz-service/engine/conformance/conformance_v2_test.go]: Please use constant so the definition is only given once per service:
s/"(unassigned)"/constants.UnassignedProjectID/

File: components/authz-service/engine/conformance/conformance_v2_test.go [hunk 5]
```diff
@@ -362,7 +359,7 @@ func TestV2ProjectsAuthorized(t *testing.T) {
 			})
 
 			t.Run("policy that allows all projects returns all when all projects requested", func(t *testing.T) {
-				proj3, proj4 := "proj-3", "proj-4"
+				proj3, proj4, unassigned := "proj-3", "proj-4", "(unassigned)"
```
Comment (msorens) [inline components/authz-service/engine/conformance/conformance_v2_test.go]: This line is a phantom; delete it please.

File: components/authz-service/server/v2/system.go (hunk 3 above)
Comment (blakestier) [reply components/authz-service/server/v2/system.go]: We can't manage deletion of specific members of a policy with a policy (only general membership).  We'd need to modify the API. We could make a card for that if we think it's worth it.

File: components/authz-service/engine/conformance/conformance_v2_test.go (hunk 5 above)
Comment (blakestier) [reply components/authz-service/engine/conformance/conformance_v2_test.go]: nice catch!

**Pull request merged**